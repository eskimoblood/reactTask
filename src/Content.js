import React, { PropTypes } from 'react';

function Content(props) {
  switch (props.state) {
    case 'NOT_ASKED':
      return <p>please enter a user name</p>;
    case 'LOADING':
      return <p>Loading</p>;
    case 'ERROR':
      return <p>{props.error}</p>;
    case 'SUCCESS':
      return (
        <ul>
          {props.data.map((repo) =>
            <li key={repo.id}><a href={repo.url}>{repo.name}</a></li>
          )}
        </ul>
      );
    default:
      return null
  }
}

Content.propTypes = {
  name: PropTypes.string,
  url: PropTypes.string,
  id: PropTypes.number
};

export default Content;
