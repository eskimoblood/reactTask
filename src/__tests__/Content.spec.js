import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Content from '../Content';

describe('Content render correct for state', () => {

  it('NOT_ASKED', () => {
    const content = shallow(<Content state="NOT_ASKED" />);
    expect(shallowToJson(content)).toMatchSnapshot();
  });

  it('LOADING', () => {
    const content = shallow(<Content state="LOADING" />);
    expect(shallowToJson(content)).toMatchSnapshot();
  });

  it('ERROR', () => {
    const content = shallow(<Content state="ERROR" error="Some strange things happen" />);
    expect(shallowToJson(content)).toMatchSnapshot();
  });

  it('SUCCESS', () => {
    const data = [
      { id: 1, url: 'http://test.de/1', name: 'repo1' },
      { id: 2, url: 'http://test.de/2', name: 'repo2' },
      { id: 3, url: 'http://test.de/3', name: 'repo3' }
    ];
    const content = shallow(<Content state="SUCCESS" data={data} />);
    expect(shallowToJson(content)).toMatchSnapshot();
  });

});
