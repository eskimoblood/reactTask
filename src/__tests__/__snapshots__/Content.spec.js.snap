exports[`Content render correct for state ERROR 1`] = `
<p>
  Some strange things happen
</p>
`;

exports[`Content render correct for state LOADING 1`] = `
<p>
  Loading
</p>
`;

exports[`Content render correct for state NOT_ASKED 1`] = `
<p>
  please enter a user name
</p>
`;

exports[`Content render correct for state SUCCESS 1`] = `
<ul>
  <li>
    <a
      href="http://test.de/1">
      repo1
    </a>
  </li>
  <li>
    <a
      href="http://test.de/2">
      repo2
    </a>
  </li>
  <li>
    <a
      href="http://test.de/3">
      repo3
    </a>
  </li>
</ul>
`;
